tests/serverspec/es_v5_spec.rb

```diff
@@ -0,0 +1,159 @@
+  es_plugin_command = "/usr/local/bin/elasticsearch-plugin"
+  es_plugins_directory = "/usr/local/lib/elasticsearch/plugins"
+  es_data_directory = "/var/db/elasticsearch"
+when "openbsd"
+  default_group = "wheel"
+  es_user_name = "_elasticsearch"
+  es_user_group = "_elasticsearch"
+  es_plugin_command = "/usr/local/elasticsearch/bin/plugin"
+  es_plugins_directory = "/usr/local/elasticsearch/plugins"
+  es_data_directory = "/var/elasticsearch"
+end
+
+describe file(es_data_directory) do
+  it { should be_directory }
+  it { should be_owned_by es_user_name }
+  it { should be_grouped_into es_user_group }
+  it { should be_mode 750 }
+end
+
+describe file(es_log_directory) do
+  it { should be_directory }
+  it { should be_owned_by es_user_name }
+  it { should be_grouped_into es_user_group }
+  it { should be_mode 750 }
+end
+
+describe service(es_service_name) do
+  it { should be_running }
+end
+
+describe package(es_package_name) do
+  it { should be_installed }
+end
+
+case os[:family]
+when "freebsd"
+  describe file("/etc/rc.conf.d") do
+    it { should be_directory }
+    it { should be_mode 755 }
+    it { should be_owned_by default_user }
+    it { should be_grouped_into default_group }
+  end
+
+  describe file("/etc/rc.conf.d/elasticsearch") do
+    it { should be_file }
+    it { should be_mode 644 }
+    it { should be_owned_by default_user }
+    it { should be_grouped_into default_group }
+    its(:content) { should match(Regexp.escape('ES_JAVA_OPTS="-XX:+UseCompressedOops -Xms1g -Xmx1g"')) }
+  end
+
+  #  XXX `process` does not support FreeBSD's `ps(1)`
+  #
+  #  describe process("/usr/local/openjdk8/bin/java") do
+  #    it { should be_running }
+  #    its(:args) { should match(Regexp.escape("-XX:+UseCompressedOops")) }
+  #  end
+  describe command("ps axww") do
+    its(:stdout) { should match(/#{ Regexp.escape("/usr/local/openjdk8/bin/java") }\s+.*#{ Regexp.escape("-XX:+UseCompressedOops") }/) }
+  end
+when "ubuntu"
+  describe file("/etc/default/elasticsearch") do
+    it { should be_file }
+    it { should be_mode 644 }
+    it { should be_owned_by default_user }
+    it { should be_grouped_into default_group }
+    its(:content) { should match(/^ES_JAVA_OPTS=\"#{ Regexp.escape("-XX:+UseCompressedOops") }\"$/) }
+  end
+
+  describe process("java") do
+    it { should be_running }
+    its(:args) { should match(Regexp.escape("-XX:+UseCompressedOops")) }
+  end
+when "redhat"
+  describe file("/etc/sysconfig/elasticsearch") do
+    it { should be_file }
+    it { should be_mode 644 }
+    it { should be_owned_by default_user }
+    it { should be_grouped_into default_group }
+    its(:content) { should match(/^ES_JAVA_OPTS=\"#{ Regexp.escape("-XX:+UseCompressedOops") }\"$/) }
+  end
+
+  describe process("java") do
+    it { should be_running }
+    its(:args) { should match(Regexp.escape("-XX:+UseCompressedOops")) }
+  end
+when "openbsd"
+  describe file("/etc/elasticsearch/jvm.in") do
```

Comment:
true, the spec file is a mess/. but i just do not want to spend time on hackish version. es 5.x is manageable in this regard, in which these Java options are in one place, `jvm.options`.

Reply:
it seems the spec file is tested on FreeBSD only according to the `.kitchen.yml` and the specs for the other platforms are not necessary.

Reply:
that is because i have not studied changes, updates, and bugs in 5.x. no conclusion yet, but the role will either support 5.x and 2.x, drop 2.x support, or support 2.x only. i need this spec for that.

Reply:
i still do not understand why tests for 2.x are included in the spec file for 5.x.
`tests/serverspec/default_spec.rb` is for 2.x and all the platforms.
`tests/serverspec/es_v5_spec.rb` is for 5.x and FreeBSD only.
the tests in `es_v5_spec.rb` for the other platforms are written for 2.x and not executed.
are they just reserving spaces for the future support?
they are harmless but confusing.

Reply:
ab07e70 makes the PR clear.